cni/pkg/repair/repair.go

```diff
@@ -145,6 +155,7 @@ func (bpr BrokenPodReconciler) deleteBrokenPod(pod v1.Pod) error {
 		return nil
 	}
 	log.Infof("Pod detected as broken, deleting: %s/%s", pod.Namespace, pod.Name)
+	defer bpr.Metrics.PodsRepaired.Increment()
```

Comment:
Why defer here?

Reply:
@stewartbutler can you clarify?

Reply:
I only used defer here to avoid using a temp variable for the return value of the delete operation.

Reply:
I don't follow, what is the temp value you are referring to?

Reply:
Please ignore that. I was just being an idiot. @therealmitchconnors you can remove the defer. 

(My insomnia-addled rationale was 'If I defer this, it will increment the metric only if the call succeeds and I don't have to create a temp variable to store the return value of the repair call', but with a defer it'd be called even if the repair called failed so there is no reason not to just increment it inline. Just put it down to me being dumb.)

Reply:
If you wanted to do this properly you could store the return of the Delete operation, and only increment if err == null.

Reply:
shouldn't we add a bit more granularity here? Perhaps record the successful and failed repairs. I am assuming this records all attempts at the pod repairs/deletes